File: api/employment/definitions/employment.ts
export interface Job {
    name: string;
    label: string;
    grades: Record<number, JobGrade>;
}

export interface PlayerJob {
    name: string;
    label: string;
    grade: JobGrade & { rank: number };
}

export interface JobGrade {
    name: string;
    label: string;
    salary: number;
    isBoss: boolean;
}

export interface ClientEmploymentProvider {
    getPlayerJob(): PlayerJob | undefined;
    playerHasJob(jobName: string, jobGrade?: number): boolean;
    onPlayerJobChanged(callback: (job: PlayerJob) => void): { unsubscribe: () => void };
}

export interface ServerEmploymentProvider {
    getJob(jobName: string): Job | undefined;
    jobExists(jobName: string, jobGrade?: number): boolean;
    getOnlineJobCount(jobName: string): number;
    getPlayerJob(playerId: number): PlayerJob | undefined;
    setPlayerJob(playerId: number, jobName: string, jobGrade: number): boolean;
    playerHasJob(playerId: number, jobName: string, jobGrade?: number): boolean;
}

File: api/employment/definitions/employment.lua
---@class Job
---@field name string
---@field label string
---@field grades table<integer, JobGrade>

---@class PlayerJob
---@field name string
---@field label string
---@field grade PlayerJobGradeData

---@class JobGrade
---@field name string
---@field label string
---@field salary number
---@field isBoss boolean

---@class PlayerJobGradeData : JobGrade
---@field rank integer

---@class ServerEmploymentProvider
---@field getJob fun(jobName: string): Job | nil
---@field jobExists fun(jobName: string, jobGrade?: integer): boolean
---@field getOnlineJobCount fun(jobName: string): integer
---@field getPlayerJob fun(playerId: integer): PlayerJob | nil
---@field setPlayerJob fun(playerId: integer, jobName: string, jobGrade: integer): boolean
---@field playerHasJob fun(playerId: integer, jobName: string, jobGrade: integer): boolean
